### ISSUE TITLE
feat: advance payment link to propal/command and use as invoice prepaid

### ISSUE THREAD

**Issue by Slordef** [Dolibarr/dolibarr#30438]
### Feature Request

Having not found an issue for this:

The idea is to be able to create a payment entry and be able to link it to a propal or client order.

Regarding this payment, it should appear in the tier as a payment advance if it is not linked to an invoicing.

When invoicing, it should be possible to select in the payment advances to declare the part already paid.

### Use case

In France, advance payment is made frequently and is not recognized as a deposit on the comptability.

Knowing that this is a recurring thing, it is urgent to implement it

### Suggested steps

For now, payment can be created directly, but:
- We should be able to attach it to a propal or an order
- We should be able to use as a payment in invoice
- We should be able to see it on tier as advance payment

**Comment by Slordef** [comment]
I offer to manage this feature if it seems like something relevant to you

**Comment by sonikf** [comment]
Hi @Slordef 
This a great idea and a missing feature actually, your contribution will be very much appreciated

**Comment by Slordef** [comment]
Should I work from the 19 branch or develop as mentioned ?
if it can go quick in prod ?

**Comment by PsyCrow-code** [comment]
Hi @Slordef What would be the difference with 'Down payment invoice'?
![image](https://github.com/user-attachments/assets/399bef04-7bc3-428d-a9b2-3631ac3c38ba)

The function you describe seems to be like 'Down payment invoice'

**Comment by PsyCrow-code** [comment]
When you create a new bill; there is a discount links (relative and absolute)
![image](https://github.com/user-attachments/assets/8820f385-d8aa-4000-9dd3-512f9b65dde9)
that can be used as you describe

> ### Feature Request
> Having not found an issue for this:
> 
> The idea is to be able to create a payment entry and be able to link it to a propal or client order.
> 
> Regarding this payment, it should appear in the tier as a payment advance if it is not linked to an invoicing.
> 
> When invoicing, it should be possible to select in the payment advances to declare the part already paid.
> 
> ### Use case
> In France, advance payment is made frequently and is not recognized as a deposit on the comptability.
> 
> Knowing that this is a recurring thing, it is urgent to implement it
> 
> ### Suggested implementation
> _No response_
> 
> ### Suggested steps
> For now, payment can be created directly, but:
> 
> * We should be able to attach it to a propal or an order
> * We should be able to use as a payment in invoice
> * We should be able to see it on tier as advance payment

**Comment by hasse-basse** [comment]
But in France you must issue a "Facture d'acompte", prepayment invoice. The payment cannot be linked just to a price proposal or to an order, and the prepayment invoice must be in the same number series ans the normal invoices. Furthermore, since January 2023, VAT must be mentioned.

**Comment by PsyCrow-code** [comment]
Maybe I don't fully understand the feature requested; That's why I think the feature request, is already built into Dolibarr

**Comment by sonikf** [comment]
Ηι
For other countries like Greece down payment invoice is not allowed, its just a  payment.
So in many cases after client accepts the propal and makes a down payment there is no way to record  it because there is no invoice yet.

**Comment by sonikf** [comment]
> Should I work from the 19 branch or develop as mentioned ?

New features always in develop, that means your target is 21!

**Comment by Slordef** [comment]
> Hi @Slordef What would be the difference with 'Down payment invoice'? ![image](https://private-user-images.githubusercontent.com/93346975/351416890-399bef04-7bc3-428d-a9b2-3631ac3c38ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3OTk4MjAsIm5iZiI6MTcyMTc5OTUyMCwicGF0aCI6Ii85MzM0Njk3NS8zNTE0MTY4OTAtMzk5YmVmMDQtN2JjMy00MjhkLWE5YjItMzYzMWFjM2MzOGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI0VDA1Mzg0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4OTQwNDllZjIzMzZjMDhiNjk2ZWI4NDc4ZDk4Mzg4OTBlMjZlOWQyNGM2OGY0Yzk4ZDliNDVmNmRhOTNmMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5dA9VPYNh9IyeDNDwU2WA44uxjx3LcgejR3uRYnXO30)
> 
> The function you describe seems to be like 'Down payment invoice'

Precisely, it is not a Down Payment Invoice, since it is not linked to invoicing which must be done afterwards. And from an accounting point of view, this cannot be counted as a down payment.

It's simply an Advance on payment. Which can be used as a prepaid part on invoicing

**Comment by PsyCrow-code** [comment]
So that could be a discount.

**Comment by Slordef** [comment]
It's not too

**Comment by Slordef** [comment]
as describe :
it's a advance on payment with no invoicing.
Invoice is made months after and use this payment as a part of already paid

So it's not a down payment, it's not a discount, and it can't be an invoicing.

For more information : https://www.service-public.fr/particuliers/vosdroits/F31187 and see "Avance" and the diff with a invoicing or a down payment

**Comment by Slordef** [comment]
@Amael-PE if you want add more information

**Comment by Amael-PE** [comment]
Re opening the topic, @PsyCrow-code , is is almost a down payment, **but** it can't be shown as a line on the invoice.
It must appear as a payment.

The issue is that when you use down payment currently + discount on the invoice later, you do have a standard invoice line on the later invoice. We need to see it as if it was only an already received payment ('cause, actually, it is one....) with the date of the received payment.
Does it makes sense?

**Comment by sonikf** [comment]
still relevant
